Corona-Warn-App/src/deviceForTesters/java/de/rki/coronawarnapp/test/organiser/ui/TestTraceLocationsViewModel.kt

```diff
@@ -0,0 +1,145 @@
+package de.rki.coronawarnapp.test.organiser.ui
+
+import androidx.lifecycle.LiveData
+import androidx.lifecycle.asLiveData
+import dagger.assisted.AssistedFactory
+import dagger.assisted.AssistedInject
+import de.rki.coronawarnapp.eventregistration.checkins.qrcode.TraceLocation
+import de.rki.coronawarnapp.eventregistration.storage.repo.TraceLocationRepository
+import de.rki.coronawarnapp.server.protocols.internal.pt.TraceLocationOuterClass
+import de.rki.coronawarnapp.test.organiser.ui.items.TestTraceLocationVH
+import de.rki.coronawarnapp.test.organiser.ui.items.TestTraceLocationItem
+import de.rki.coronawarnapp.util.TimeStamper
+import de.rki.coronawarnapp.util.coroutine.DispatcherProvider
+import de.rki.coronawarnapp.util.viewmodel.CWAViewModel
+import de.rki.coronawarnapp.util.viewmodel.SimpleCWAViewModelFactory
+import kotlinx.coroutines.flow.map
+import okio.ByteString.Companion.toByteString
+import org.joda.time.DateTime
+import java.util.UUID
+
+class TestTraceLocationsViewModel @AssistedInject constructor(
+    dispatcherProvider: DispatcherProvider,
+    private val traceLocationRepository: TraceLocationRepository,
+    private val timeStamper: TimeStamper
+) : CWAViewModel(dispatcherProvider = dispatcherProvider) {
+
+    val traceLocations: LiveData<List<TestTraceLocationItem>> = traceLocationRepository.allTraceLocations
+        .map { traceLocations ->
+            traceLocations
+                .filter { it.endDate == null || it.endDate.isAfter(timeStamper.nowUTC.toDateTime().minusDays(15)) }
+                .sortedBy { it.description }
+        }
+        .map { traceLocations ->
+            traceLocations.map { traceLocation ->
+                TestTraceLocationVH.Item(
+                    traceLocation = traceLocation,
+                    onCheckIn = { /* TODO */ },
+                    onDuplicate = { /* TODO */ },
+                    onShowPrint = { /* TODO */ },
+                    onClearItem = { traceLocationRepository.deleteTraceLocation(it) }
```

Comment:
shouldn't we show confirmation dialog before deleting intem?

Reply:
Good catch! 👍  Forgot about it.